experimental/Schemes/IdealSheaves.jl

```diff
@@ -0,0 +1,325 @@
+export IdealSheaf
+
+export scheme, covering, getindex, subscheme, covered_patches, extend!, ideal_dict
+
+export ideal_sheaf_type
+
+### Forwarding the presheaf functionality
+underlying_presheaf(I::IdealSheaf) = I.I
+
+@Markdown.doc """
+    IdealSheaf(X::ProjectiveScheme, g::Vector{<:RingElem})
+
+Create the ideal sheaf on the covered scheme of ``X`` which is 
+generated by the dehomogenization of the homogeneous elements in `g` 
+in every chart.
+"""
+function IdealSheaf(X::ProjectiveScheme, I::MPolyIdeal) 
+  S = base_ring(I)
+  S === ambient_ring(X) || error("ideal does not live in the ambient ring of the scheme")
+  g = gens(I)
+  X_covered = covered_scheme(X)
+  C = default_covering(X_covered)
+  r = fiber_dimension(X)
+  I = IdDict{AbsSpec, Ideal}()
+  for i in 0:r
+    I[C[i+1]] = ideal(OO(C[i+1]), dehomogenize(X, i).(g))
+  end
+  return IdealSheaf(X_covered, I, check=true)
+end
+
+function IdealSheaf(
+    X::ProjectiveScheme, 
+    g::MPolyElem_dec
+  )
+  return IdealSheaf(X, [g])
+end
+
+function IdealSheaf(
+    X::ProjectiveScheme, 
+    g::Vector{RingElemType}
+  ) where {RingElemType<:MPolyElem_dec}
+  X_covered = covered_scheme(X)
+  r = fiber_dimension(X)
+  I = IdDict{AbsSpec, Ideal}()
+  U = basic_patches(default_covering(X_covered))
+  for i in 1:length(U)
+    I[U[i]] = ideal(OO(U[i]), dehomogenize(X, i-1).(g))
+  end
+  return IdealSheaf(X_covered, I, check=false)
+end
+
+# this constructs the empty ideal sheaf
+function IdealSheaf(X::CoveredScheme) 
+  C = default_covering(X)
+  I = IdDict{AbsSpec, Ideal}()
+  for U in basic_patches(C)
+    I[U] = ideal(OO(U), elem_type(OO(U))[])
+  end
+  return IdealSheaf(X, I, check=false)
+end
+
+# internal routine to set up an ideal sheaf by automatic extension 
+# from one prescribed set of generators on one affine patch
+function IdealSheaf(X::CoveredScheme, U::AbsSpec, g::Vector{RET}) where {RET<:RingElem}
```

Comment:
Since `IdealSheaf` is exported this method is not internal but part of the public interface. 
So this needs to be properly documented.